web/src/main/java/org/cbioportal/web/StudyController.java

```diff
@@ -190,4 +195,8 @@ private void warmDefaultResponseCache() {
 
         return new ResponseEntity<>(cancerStudyTags, HttpStatus.OK);
     }
+    
+    private String getAccessLevel() {
+        return usingAuth() && showUnauthorizedStudiesOnHomePage ? "list" : "read";
```

Comment:
You have these string constants in a few places. Any reason why you didn't make them fully fledged constants that we can reference?

Reply:
I created constants here. Affected a lot of classes. Also, I moved the CancerStudyPermissionEvaluator from `utils` to new module `security/permission-evaluator`. The reason for this is that to be a useful util lib the `utils` module is not allowed to have any dependencies itself.